lib/msf/core/exploit/powershell.rb

```diff
@@ -224,7 +224,7 @@ def run_hidden_psh(ps_code, payload_arch, encoded)
   def cmd_psh_payload(pay, payload_arch, opts = {})
     %i[persist prepend_sleep exec_in_place encode_final_payload encode_inner_payload
     remove_comspec noninteractive wrap_double_quotes no_equals method].map do |opt|
-      opts[opt] ||= datastore["Powershell::#{opt}"]
```

Comment:
What about `opts[opt] || datastore["Powershell::#{opt}"]`?

Reply:
Wouldn't that still override the option passed to `cmd_psh_payload` if the datastore option is true by default?

Reply:
Did I mean `datastore["Powershell::#{opt}"] || opts[opt]`?

Reply:
I don't believe that will fix it either unless I'm misunderstanding.

Reply:
I don't think my brain was booted yet this morning, so let me give this a test and see if this can be refactored or not.

Reply:
I wasn't thinking through this clearly. We want to distinguish between `nil` and `false` here, so we shouldn't do a truthy/falsey evaluation. What you have is correct.

Reply:
tl;dr I made the same mistake that the author made while thinking through this. `false` is significant!

Reply:
https://github.com/rapid7/metasploit-framework/commit/a57923ee648672626855765c3a9578aab016a83c